physionet-django/project/projectfiles/gcs.py

```diff
@@ -117,6 +118,14 @@ def publish_initial(self, active_project, published_project):
 
     def publish_complete(self, active_project, published_project):
         self.rm_dir(active_project.file_root())
+        if settings.DEFAULT_PROJECT_ACCESS_MECHANISM == 'research_environment':
```

Comment:
In other places you use 
```
DataSource.AccessMechanism.RESEARCH_ENVIRONMENT
```

Reply:
I'd extract the logic of creating a default `DataSource` to a separate class that would be configured during the app's initialization with the default source. Then, after the project is published, some generic `#create_data_source` method would be called from the publishing code. This way the logic is nicely separated and not embedded everywhere.

Also - I don't think this is the place to create a `DataSource`. This should probably happen inside the `publish` method in a project (or similar). This part of the code is simply about publishing files. It shouldn't care that an abstraction like `DataSource` exists or not.